### PR TITLE
Cleanup of `JesBackendSpec`. Closes #1026

### LINKED ISSUE #1026: Assess/Fix/Implement Ignored Tests in JesBackendSpec
The goal here is to look at the tests in the spec, and determine
1. does the test assess something valuable
2. if no, delete it (with an explanation in the PR)
3. if yes, either fix it up so it passes, or reimplement... whatever is more expedient

### PR DESCRIPTION
**Pull request by kshakir**
Additionally:
Ignoring `PostMVP` specs to allow `sbt alltests` to run.
Shutting down many more actor systems by creating a `TestKitSuite` mixing Akka's `TestKit` and ScalaTest's `Suite`.
DRYed out some backend specs with a `BackendSpec`.
Moved more classes/files from `engine` to `core`, including `SampleWdl`, `application.conf`, etc.
Gave more time to the integration test `SprayDockerRegistryApiClientSpec`.
`DockerTest` and `IntegrationTest` tags now in `crowell.core`.

### CODE REVIEW

**Comment by Horneth** [comment]
I actually haven't had the time to look at this today and I'm out until Tuesday. So if anyone wants to replace me feel free, if not I'll review on Tuesday or when/if I have the chance this week-end.

**Comment by mcovarr** [comment]
:+1:

File: engine/src/test/scala/cromwell/CallCachingWorkflowSpec.scala [hunk 1]
```diff
@@ -30,7 +30,7 @@ class CallCachingWorkflowSpec extends CromwellTestkitSpec {
   )
 
   "A workflow which is run twice" should {
-    "use cached calls on the second run" taggedAs PostMVP in {
+    "use cached calls on the second run" taggedAs PostMVP ignore {
```
Comment (mcovarr) [inline engine/src/test/scala/cromwell/CallCachingWorkflowSpec.scala]: looks like you and @cjllanwarne thought alike here #1093

**Comment by scottfrazer** [comment]
👍

**Pull request merged**